scripts/us_fda/drugs/create_enums.py

```diff
@@ -0,0 +1,453 @@
+description: "An Abbreviated New Drug Application (ANDA) contains data that, when submitted to FDA's Center for Drug Evaluation and Research, Office of Generic Drugs, provides for the review and ultimate approval of a generic drug product."
+descriptionURL: "https://www.fda.gov/drugs/drug-approvals-and-databases/drugsfda-glossary-terms#ANDA"
+
+Node: dcid:ApplicationTypeBLA
+typeOf: ApplicationTypeEnum
+name: "Biologic License Application (BLA)"
+description: "A biologics license application is a submission that contains specific information on the manufacturing processes, chemistry, pharmacology, clinical pharmacology and the medical affects of the biologic product."
+descriptionURL: "https://www.fda.gov/drugs/drug-approvals-and-databases/drugsfda-glossary-terms#BLA"
+'''
+
+
+# SCRIPT: Create Enum Files and associated Dictionaries
+def generate_enums(should_write_file):
+    """Generates enum schema file and returns loaded enum dictionaries
+
+    If should_write_file is true, then all enums are written to
+    FDADrugsEnumSchema.mcf .
+    The loaded dictioanries are always returned. These dictionaries are used in
+    clean_data.py and write_mcf.py .
+    """
+
+    if should_write_file:
+        enums_schema_file = open('FDADrugsEnumSchema.mcf', 'w')
+        enums_schema_file.write(ENUMS_HANDWRITTEN)
+    else:
+        enums_schema_file = None
+
+    # create dosage_form enum dict and write dosage_form Enum Schema to file
+    dosage_forms_fda = pd.read_csv('raw_data/FDADosageForms.csv')
+    dosage_form_to_dcid = {}
+    dosage_forms_fda.apply(lambda x: parse_dosage_form_fda(
+        enums_schema_file, dosage_form_to_dcid, x),
+                           axis=1)
+    add_custom_dosage_form_enums(enums_schema_file, dosage_form_to_dcid)
+
+    # create adminROute enum dict and write admin_route Enum Schema to file
+    admin_route_to_dcid = {}
+    admin_route_fda = pd.read_csv('raw_data/FDAAdminRoutes.csv')
+    admin_route_fda.apply(lambda x: parse_admin_route_fda(
+        enums_schema_file, admin_route_to_dcid, x),
+                          axis=1)
+    add_custom_admin_route_enums(enums_schema_file, admin_route_to_dcid)
+
+    # create applType enum dict
+    appl_type_to_dcid = {
+        'NDA': 'ApplicationTypeNDA',
+        'ANDA': 'ApplicationTypeANDA',
+        'BLA': 'ApplicationTypeBLA'
+    }
+
+    # create TherapueticEquivalence Enum Schema File + dictionary
+    te_code_to_dcid = {}
+    create_te_code_enums(enums_schema_file, te_code_to_dcid)
+
+    # create Marketing Status Dictionary
+    market_stat_to_dcid = {
+        '1': 'MarketingStatusPrescription',
+        '2': 'MarketingStatusOverTheCounter',
+        '3': 'MarketingStatusDiscontinued',
+        '4': 'MarketingStatusNone'
+    }
+
+    if enums_schema_file:
+        enums_schema_file.close()
+    return dosage_form_to_dcid, admin_route_to_dcid, appl_type_to_dcid, te_code_to_dcid, market_stat_to_dcid
+
+
+def main():
```

Comment:
Its good practice to use app/flags from absl. Esp for passing flags like --write_enums_to_file.

For example: https://github.com/datacommonsorg/data/blob/master/scripts/proteinInteractionEBI/parse_ebi.py#L23-L24